packages/runtime/container-runtime/src/blobManager/blobManager.ts

```diff
@@ -280,21 +276,33 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 					expired,
 				});
 				if (expired) {
-					// we want to avoid submitting ops with broken handles
-					this.closeContainer(
-						new GenericError("Trying to submit a BlobAttach for expired blob", undefined, {
-							localId,
-							blobId,
-							secondsSinceUpload,
-						}),
-					);
+					// handle reupload asynchronously
+					this.reuploadBlobAndSendOp(localId, pendingEntry);
+					return; // Return void to avoid promise violation
 				}
 			}
 			pendingEntry.opsent = true;
 			return sendBlobAttachOp(localId, blobId);
 		};
 	}
 
+	private reuploadBlobAndSendOp(localId: string, pendingEntry: PendingBlob): void {
+		(async () => {
+			await this.uploadBlob(localId, pendingEntry.blob).then(() =>
+				// uploadBlob and onUploadResolve will not send another blob attach so we need to do it here
+				this.sendBlobAttachOp(localId, pendingEntry.storageId),
+			);
+			// Once the blob is re-uploaded, proceed with sending the BlobAttachOp
+			pendingEntry.opsent = true;
+		})().catch((error) => {
```

Comment:
What is the purpose of defining a function here? Could we just use try/catch instead of () function invocation? Code looks clean that way and also avoids logging error stack with an `anonymous` function name.

Reply:
I was experimenting a bit with the function since I was having issues with the async call. I just simplified it. Thanks.